src/mono/sample/iOS/symbols_to_keep.list

```diff
@@ -0,0 +1,111 @@
+_SystemNative_Abort
+_SystemNative_Exit
+_SystemNative_SchedGetCpu
+_SystemNative_ConvertErrorPlatformToPal
+_SystemNative_ConvertErrorPalToPlatform
+_SystemNative_StrErrorR
+_SystemNative_GetLoadLibraryError
+_SystemNative_GetProcAddress
+_SystemNative_FreeLibrary
+_SystemNative_GetErrNo
+_SystemNative_SetErrNo
+_SystemNative_GetEnviron
+_SystemNative_FreeEnviron
+_SystemNative_GetOSArchitecture
+_SystemNative_GetNonCryptographicallySecureRandomBytes
+_SystemNative_GetCryptographicallySecureRandomBytes
+_SystemNative_GetSystemTimeAsTicks
+_SystemNative_GetTimestamp
+_SystemNative_LChflagsCanSetHiddenFlag
+_SystemNative_CanGetHiddenFlag
+_SystemNative_Log
+_SystemNative_LogError
+_SystemNative_LowLevelMonitor_Create
+_SystemNative_LowLevelMonitor_Destroy
+_SystemNative_LowLevelMonitor_Acquire
+_SystemNative_LowLevelMonitor_Release
+_SystemNative_LowLevelMonitor_Wait
+_SystemNative_LowLevelMonitor_Signal_Release
+_SystemNative_AlignedAlloc
+_SystemNative_AlignedFree
+_SystemNative_AlignedRealloc
+_SystemNative_Calloc
+_SystemNative_Free
+_SystemNative_Malloc
+_SystemNative_Realloc
+_SystemNative_GetAllMountPoints
+_SystemNative_GetReadDirRBufferSize
+_SystemNative_ReadDirR
+_SystemNative_GetEUid
+_SystemNative_GetEGid
+_SystemNative_GetPid
+_SystemNative_CreateAutoreleasePool
+_SystemNative_DrainAutoreleasePool
+_SystemNative_CreateThread
+_SystemNative_Access
+_SystemNative_ChDir
+_SystemNative_ChMod
+_SystemNative_FChMod
+_SystemNative_Close
+_SystemNative_CopyFile
+_SystemNative_LoadLibrary
+_SystemNative_GetDefaultSearchOrderPseudoHandle
+_SystemNative_GetFileSystemType
+_SystemNative_FLock
+_SystemNative_FSync
+_SystemNative_FTruncate
+_SystemNative_GetCpuUtilization
+_SystemNative_GetCwd
+_SystemNative_GetDefaultTimeZone
+_SystemNative_GetEnv
+_SystemNative_GetHostName
+_SystemNative_GetProcessPath
+_SystemNative_GetUnixVersion
+_SystemNative_GetUnixRelease
+_SystemNative_LChflags
+_SystemNative_FChflags
+_SystemNative_Link
+_SystemNative_LockFileRegion
+_SystemNative_LowLevelMonitor_TimedWait
+_SystemNative_LSeek
+_SystemNative_MkDir
+_SystemNative_MkdTemp
+_SystemNative_MksTemps
+_SystemNative_Open
+_SystemNative_PathConf
+_SystemNative_PosixFAdvise
+_SystemNative_FAllocate
+_SystemNative_PRead
+_SystemNative_PReadV
+_SystemNative_PWrite
+_SystemNative_PWriteV
+_SystemNative_Read
+_SystemNative_OpenDir
+_SystemNative_CloseDir
+_SystemNative_ReadLink
+_SystemNative_Rename
+_SystemNative_RmDir
+_SystemNative_FStat
+_SystemNative_Stat
+_SystemNative_LStat
+_SystemNative_SymLink
+_SystemNative_SysConf
+_SystemNative_SysLog
+_SystemNative_Unlink
+_SystemNative_UTimensat
+_SystemNative_FUTimens
+_SystemNative_Write
+_SystemNative_GetGroups
+_SystemNative_GetPwUidR
+_SystemNative_GetPwNamR
+_SystemNative_SearchPath
+_SystemNative_SearchPath_TempDirectory
+_SystemNative_FcntlSetFD
+_SystemNative_MMap
+_SystemNative_MUnmap
+_SystemNative_MSync
+_SystemNative_MAdvise
+_SystemNative_ShmOpen
+_SystemNative_ShmUnlink
```

Comment:
How were these symbols obtained and where are they coming from?

Reply:
These symbols were obtained by stripping out all the symbols except required ones for normal application execution. Assumption was that only `_ios_register_button_click` and `_ios_set_text` global symbols are required. After test run, the following stack trace is obtained on `iossimulator` when keeping only the two mentioned symbols.
```
[14:53:20.6938730]   at Xamarin.Hosting.Services.LaunchApplicationInSimulatorInternalAsyncSimCtl (Xamarin.Hosting.SimulatorLaunchConfig config) [0x00555] in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Hosting/Services.cs:1053 
[14:53:20.6939100]   at Xamarin.Hosting.Services.LaunchApplicationInSimulatorAsync (Xamarin.Hosting.SimulatorLaunchConfig config) [0x0038c] in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Hosting/Services.cs:1171 
[14:53:20.6939150]   at Xamarin.Launcher.SimController.LaunchAsync (Xamarin.Hosting.SimulatorLaunchConfig config) [0x000de] in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/controller-simulator.cs:689 
[14:53:20.6939180]   at Xamarin.Launcher.Driver.LaunchSimulatorAsync () [0x004a7] in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/Main.cs:808 
[14:53:20.6939430]   at Xamarin.Utils.NSRunLoopExtensions.RunUntilTaskCompletion[T] (Foundation.NSRunLoop this, System.Threading.Tasks.Task`1[TResult] task) [0x00082] in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Utils/Extensions.cs:35 
[14:53:20.6939590]   at Xamarin.Launcher.Driver.Main2 (System.String[] args) [0x00ba3] in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/Main.cs:435 
[14:53:20.6939620]   at Xamarin.Launcher.Driver.Main (System.String[] args) [0x0006d] in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/Main.cs:123 
[14:53:20.7041950] Process mlaunch exited with 1
```
Conclusion was that the execution relies on Xamarin launch framework and thus require the same set or a subset of symbols as for a Xamarin run. 

The difference in size when only `_ios_register_button_click` and `_ios_set_text` are kept after stripping, compared to the full set of symbols specified in `symbols_to_keep.list` file is 4600 bytes.

Reply:
I was more referring to "how did we figure out this exact list of symbols is needed?", as the PR does not describe how the `symbols_to_keep.list` was obtained.
Can we just use that approach 'as is' or do we need to implement it?

Reply:
`symbols_to_keep.list` is obtained manually and it works for this specific application only. It could be useful to automate it as a part of a build process.

Reply:
After offline discussion, we decided to preserve global symbols and to make it available as an option in AppleAppBuilder.